packages/documentation-framework/scripts/md/parseMD.js

```diff
@@ -281,9 +281,14 @@ function writeIndex() {
 }
 
 // Build unique names for components if they are a part of the "next" module.
-function getTsDocName(name, isNextComponent) {
-  return `${name}${isNextComponent ? '-next' : ''}`;
-};
+function getTsDocName(name, isNextComponent, isDeprecatedComponent) {
```

Comment:
So as to not have to add new parameters to this function whenever a new condition arises, we might want to consider reworking this a bit. This is what I had in mind roughly, where if there's another "variant" at any point, we just have to update the 1 location in `getTsDocNameVariant`:

```
// Build unique names for components with a "variant" extension
function getTsDocName(name, variant) {
  return `${name}${variant ? `-${variant}` : ''}`;
}

function getTsDocNameVariant(source) {
  if (source.includes('next')) {
    return 'next';
  }

  if (source.includes('deprecated')) {
    return 'deprecated';
  }
}
```

and the 2 usages above would then look like the following:
`tsDocs[getTsDocName(name, getTsDocNameVariant(file))] = { name, description, props };`
and
`const name = getTsDocName(componentName, getTsDocNameVariant(source));`